### PR TITLE
DOC: add an example to `PeriodIndex.to_timestamp` to clarify the behavior in case len(index)<3

### PR DESCRIPTION
**Pull request by natmokval**
xref #56213

added an example to `PeriodIndex.to_timestamp` to clarify the behavior in case `len(index) < 3`

### CODE REVIEW

**Comment by natmokval** [comment]
@MarcoGorelli, I added an example to clarify the behavior of `PeriodIndex.to_timestamp` in case `len(index) < 3`. Could you please take a look at this PR?

File: pandas/core/arrays/period.py [hunk 1]
```diff
@@ -636,6 +636,12 @@ def to_timestamp(self, freq=None, how: str = "start") -> DatetimeArray:
         >>> idx.to_timestamp()
         DatetimeIndex(['2023-01-01', '2023-02-01', '2023-03-01'],
         dtype='datetime64[ns]', freq='MS')
+
+        We can not infer the frequency if len(index) < 3:
```
[the remark this answers is not in the document]
Comment (natmokval) [reply pandas/core/arrays/period.py]: thanks for the comment @datapythonista, I corrected docstring as you suggested. I also added a new example to emphasize importance values of index to be strictly monotonic.

File: pandas/core/arrays/period.py [hunk 2]
```diff
@@ -636,6 +636,20 @@ def to_timestamp(self, freq=None, how: str = "start") -> DatetimeArray:
         >>> idx.to_timestamp()
         DatetimeIndex(['2023-01-01', '2023-02-01', '2023-03-01'],
         dtype='datetime64[ns]', freq='MS')
+
+        The frequency will not be inferred if the index contains less than
+        three elements and values of index are not strictly monotonic:
```
Comment (MarcoGorelli) [inline pandas/core/arrays/period.py:641]: the way this is phrased it sounds like (pseudocode):
```
if len(index) < 3 and not index.is_monotonic():
    dont_infer_frequency(...)
```
Instead, I think it's more like
```
if len(index) < 3 or not index.is_monotonic():
    dont_infer_frequency(...)
```
?

How about
```
        The frequency will not be inferred if the index contains less than
        three elements, or if the values of index are not strictly monotonic:
```

**Review by MarcoGorelli — COMMENTED** [review]
nice!

File: pandas/core/arrays/period.py (hunk 2 above)
Comment (natmokval) [reply pandas/core/arrays/period.py]: thanks, you are right, my wording isn't correct. I changed it as you suggested.

**Review by MarcoGorelli — APPROVED** [review]
thanks @natmokval

**Pull request merged**